src/cpp/src/scheduler.hpp

```diff
@@ -85,6 +87,15 @@ class Scheduler {
         return m_config;
     }
 
+    void free_blocks_from_sequence(size_t seq_id, const std::set<size_t>& logical_block_indices_to_free) {
+        OPENVINO_ASSERT(m_num_layers == 0, "this overload may only be used if num_layers == 0");
+        m_block_manager.free_blocks_from_sequence(seq_id, logical_block_indices_to_free);
+    }
```

Comment:
Am I right that beam search / parallel sampling is not planned to be supported? If yes, let's insert assert during adding request to CB pipeline

Reply:
The algo itself works, but is negatively impacted by the beam search and parallel sampling cases since blocks belonging to sequences in the same group are currently evicted independently, and since they are reference-counted, this means that the prompt blocks for such generation configs have a very low (but a non-zero) chance of being completely physically evicted. We could expand the algo to beam search and parallel sampling in the upcoming PRs by imposing cross-sequence dependencies on block eviction. I don't think we need to block the path entirely by the assertion, though, since generation-phase blocks still have a reasonable chance to be evicted.